src/resolve/resolved-conf.c

```diff
@@ -385,6 +432,110 @@ int config_parse_dnssd_txt(const char *unit, const char *filename, unsigned line
         return 0;
 }
 
+int config_parse_dns_stub_listener_extra(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ DNSStubListenerExtra *udp = NULL, *tcp = NULL;
+        _cleanup_free_ char *word = NULL;
+        Manager *m = userdata;
+        bool both = false;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                m->dns_extra_stub_listeners = ordered_set_free(m->dns_extra_stub_listeners);
+                return 0;
+        }
+
+        p = rvalue;
+        r = extract_first_word(&p, &word, ":", 0);
```

Comment:
I don't understand how this is supposed to work, especially with IPv6 addresses. Please add a unit test that verifies that this parser does the right thing.

Reply:
It's bit tricky.   For example ```udp:[2001:db8:3c4d:15::1a2f:1a2b]:2201``` . First we look for `udp/tcp` here. If this successful we allocate the `udp/tcp` object. If it's missing then we pass the string  like ```db8:3c4d:15::1a2f:1a2b```  to ```socket_addr_port_from_string_auto```, which may be a ipv6 address missing the port.  `socket_addr_port_from_string_auto` will parse this .  Similarly for ```udp:172.16.85.231:2202```.  or for ```172.16.85.231:2202``` / ```172.16.85.231``` .  Yes will add example .